rustls/src/error.rs

```diff
@@ -167,6 +167,7 @@ pub enum PeerMisbehaved {
     SelectedUnofferedPsk,
     SelectedUnusableCipherSuiteForVersion,
     ServerHelloMustOfferUncompressedEcPoints,
+    ServerNameDifferedOnRetry,
```

Comment:
Using `ServerName` here vs `Sni` in `PeerMisbehaved`, should probably be consistent at least.